### ISSUE TITLE
MariaDB via socket

### ISSUE THREAD

**Issue by KB7777** [nilsteampassnet/TeamPass#1893]
Hello :)

Is it possible to connect to MariaDB via socket?
If yes where can I change this setting?

There is no such setting at /var/www/html/teampass/includes/config/settings.php if I want to change it from port to socket.

Regards.

**Comment by nilsteampassnet** [comment]
Not sure to provide a solution but can you do the next try.

Open file `/includes/config/settings.php`
Change content of variable `$server` by something like `localhost:<path_to_mysql.sock>`
With for example `<path_to_mysql.sock>` = `/var/run/mysqld/mysqld.sock`

Make a try

**Comment by wirerydr** [comment]
I was interested in this as well, and gave it a try.  Unfortunately the connection failed with the following log:
```
[Thu Sep 07 23:35:50.374354 2017] [php7:notice] [pid 557] [client 192.168.240.120:57524] PHP Warning:  mysqli_connect(): (HY000/2002): Connection refused in /opt/teampass/teampass__20170829-56a9ad9/index.php on line 87
[Thu Sep 07 23:35:50.374392 2017] [php7:notice] [pid 557] [client 192.168.240.120:57524] PHP   1. {main}() /opt/teampass/teampass__20170829-56a9ad9/index.php:0
[Thu Sep 07 23:35:50.374395 2017] [php7:notice] [pid 557] [client 192.168.240.120:57524] PHP   2. mysqli_connect() /opt/teampass/teampass__20170829-56a9ad9/index.php:87
[Thu Sep 07 23:35:50.374914 2017] [php7:notice] [pid 557] [client 192.168.240.120:57524] PHP Fatal error:  Uncaught Error: Call to a member function set_charset() on boolean in /opt/teampass/teampass__20170829-56a9ad9/index.php:88\nStack trace:\n#0 {main}\n  thrown in /o
pt/teampass/teampass__20170829-56a9ad9/index.php on line 88
```
I investigated a bit further.  Quoting from the the [official PHP documentation](https://php.net/manual/en/mysqli.construct.php) covering _mysqli_connect()_ calls:
```
host can be either a host name or an IP address. Passing the NULL value or the string "localhost" to this parameter, the local host is assumed. When possible, pipes will be used instead of the TCP/IP protocol.
```

However, the docs also said that PHP-configured defaults will be used for both the port and socket if you don't specify them in the _mysqli_connect()_ call.  This sounded promising because even without modifying the Teampass code to provide an explicit UNIX socket argument to _mysqli_connect()_, it would still try to use a default socket if it could.  This was intriguing so I tested it.

1. I reconfigured MySQL to listen on both a non-standard TCP port and a non-standard UNIX socket, and then restarted MySQL.  Without changing any PHP / Teampass / Apache settings, I tried accessing Teampass, which failed.

2. Next I used [socat](http://www.dest-unreach.org/socat/doc/socat.html) to redirect any requests intended for MySQL's original TCP port _3306_ and UNIX socket _/var/lib/mysql/mysql.sock_ to the new, non-standard port and socket I set up in step number 1.  I also had socat log all requests to separate files, one for the port and one for the socket, so I could quickly tell which method was being used..  Then (again without changing any PHP / Teampass / Apache settings) I tried accessing Teampass.  This time it worked, with the MySQL requests/responses showing up in the UNIX socket log.

What this proved to me is that as long as I specify "_localhost_" in my Teampass _includes/config/settings.php_ PHP will in-fact automatically prefer using the UNIX socket instead of the TCP port.  There doesn't seem to be any extra PHP / Teampass / Apache configuration required to do what you want.  I'd recommend confirming that this works in your case with tools like socat or perhaps tcpdump.

Hope this helps a bit...
